src/freenet/node/AnnounceSender.java

```diff
@@ -484,6 +484,15 @@ private void timedOut(PeerNode next) {
 	}
 
 	private void rnf(PeerNode next) {
+		synchronized(this) {
+			while(waitingForTransfers > 0) {
+				try {
+					wait();
+				} catch (InterruptedException e) {
+					// Ignore.
```

Comment:
Unfortunately, I don't think that I understand Freenet well enough to know what needs to be done if a thread got interrupted while waiting for remaining transfers to finish. If I understood correctly, an InterruptedException occurs only if another thread has called the interrupt-method of the AnnounceSender-thread. A typical example where this could happen would be during a shutdown of Freenet. However, this probably means that whenever this exception gets thrown, one needs to check whether it is due to a shutdown, a timeout or any other reason, as it determines whether the thread needs to terminate or if it should just continue or else. A quick grep through Freenet's code revealed that ignoring the InterruptedException seems to be common practice. Maybe this should be addressed in a separate issue report?